### PR TITLE
feat(destination-type): Prometheus Remote Write destination

### PR DESCRIPTION
**Pull request by antonblock**
### Proposed Change

This is just https://github.com/observIQ/bindplane-op/pull/579 being readded now that the issue with seeding resources with map params is resolved

* Finalized prometheus remote write destionation
* Added documentation for Prom remote write
* Update docs/www/integrations/destinations/prometheus-remote-write.md
* Update resources/destination-types/prometheus_remote_write.yaml
* Updated TLs parameters and broke up URL
* Made some PR fixups

##### Checklist
- [x] Changes are tested
- [x] CI has passed

### CODE REVIEW

**Comment by jsirianni** [comment]
Do we have control over positioning? HTTP header and external label options are not aligned. 


Screenshot from 2023-01-23 10-34-59

File: resources/destination-types/prometheus_remote_write.yaml [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+      advancedConfig: true
+
+    - name: namespace
+      label: Namespace
+      description: Prefix to attach to each metric name.
+      type: string
+      default: ""
+      advancedConfig: true
+
+    - name: enable_resource_to_telemetry_conversion
```
Comment (jsirianni) [inline resources/destination-types/prometheus_remote_write.yaml:128]: I think the UI would look nicer if this option was above enable TLS

**Comment by antonblock** [comment]
> Do we have control over positioning? HTTP header and external label options are not aligned.
> 

We don't have that level of control, but I can try to rearrange it

File: resources/destination-types/prometheus_remote_write.yaml (hunk 1 above)
Comment (antonblock) [reply resources/destination-types/prometheus_remote_write.yaml]: @jsirianni updated the arrangement here

**Review by cpheps — APPROVED** [review]

File: resources/destination-types/prometheus_remote_write.yaml [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+            {{ end }}
+          {{ end }}
+
+          {{ if ne .namespace "" }}
+          namespace: {{ .namespace }}
+          {{ end }}
+
+          {{ if .enable_write_ahead_log }}
+          wal:
+            directory: $OIQ_OTEL_COLLECTOR_HOME/storage/prom_rw
```
Comment (jsirianni) [inline resources/destination-types/prometheus_remote_write.yaml:200]: Will this cause an issue if we have multiple remote write destinations on a single system?

I configured two, and I see the following directory structure:

```
[root@fedora]/opt/observiq-otel-collector# tree storage 
storage
└── prom_rw
    └── prom_remotewrite
        └── 00000000000000000001
```

Comment (cpheps) [reply resources/destination-types/prometheus_remote_write.yaml:200]: So it appends the `prom_remotewrite` directory to whatever path is passed in [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/prometheusremotewriteexporter/wal.go#L90). We likely want make the `directory` parameter unique to each instance.

**Review by jsirianni — APPROVED** [review]

**Pull request merged**